src/symbol.cpp

```diff
@@ -65,7 +68,7 @@ void Symbol::importIndexes(std::queue<Symbol*>* list){
 
 IntervalSet* Symbol::exportIndex(){
     IntervalSet* tmp = this->range;
-    this->range = NULL;
+    this->range = NULL; // TODO: memory leak?
     delete this;
```

Comment:
The function implicitly transfers ownership of the IntervalSet instance to the caller, this can lead to memory leaks to whoever is going to use such function in the future and is not aware of this fact. 

In fact it is not clear to me whether you intend Symbol to have ownership over range or not, as this function seems to suggest that it should not. 

If that is the case, then this change that i made is likely to be wrong within your memory model:

```
     index = NULL;
 -    range = NULL;
 +    range = new IntervalSet(Interval(DNumber::minus_inf, DNumber::plus_inf));
      value = NULL;
  }
```